core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -494,6 +509,7 @@ public RetryOnReplicaException(StreamInput in) throws IOException {
         @Override
         public void onResponse(Releasable releasable) {
             try {
+                replica.updateGlobalCheckpointOnReplica(globalCheckpoint);
```

Comment:
I think it's fine as is? This code is in a try-block and I think the catch-block for it is already the right thing to do if an already closed exception is thrown?